File: app/src/main/java/ru/roscha_akademii/medialib/common/ActivityNavigator.java
package ru.roscha_akademii.medialib.common;

import android.app.Activity;

import ru.roscha_akademii.medialib.viewvideo.view.ShowVideoActivity;

public class ActivityNavigator {
    private Activity activity;

    ActivityNavigator(Activity activity) {
        this.activity = activity;
    }

    public void openVideo(long id) {
        activity.startActivity(ShowVideoActivity.getStartIntent(activity, id));
    }
}

File: app/src/main/java/ru/roscha_akademii/medialib/common/ActivityNavigator.kt
package ru.roscha_akademii.medialib.common

import android.app.Activity
import ru.roscha_akademii.medialib.book.onebook.view.OneBookActivity

import ru.roscha_akademii.medialib.video.playvideo.viewvideo.view.ShowVideoActivity

open class ActivityNavigator internal constructor(private val activity: Activity) {

    open fun openVideo(id: Long) {
        activity.startActivity(ShowVideoActivity.getStartIntent(activity, id))
    }

    open fun openBook(id: Long) {
        activity.startActivity(OneBookActivity.getStartIntent(activity, id))
    }
}
